auditbeat/module/file_integrity/kprobes/events_cache.go

```diff
@@ -0,0 +1,156 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package kprobes
+
+import (
+	"path/filepath"
+)
+
+type dEntriesIndex map[dKey]*dEntry
+type dEntriesMoveIndex map[uint64]*dEntry
+
+type dEntryCache struct {
+	index     dEntriesIndex
+	moveCache dEntriesMoveIndex
+}
+
+func newDirEntryCache() *dEntryCache {
+	return &dEntryCache{
+		index:     make(map[dKey]*dEntry),
+		moveCache: make(map[uint64]*dEntry),
+	}
+}
+
+// Get returns the dEntry associated with the given key.
+func (d *dEntryCache) Get(key dKey) *dEntry {
+	entry, exists := d.index[key]
+	if !exists {
```

Comment:
what happens if this misses? should we log something?

Reply:
we could log something for sure, but given that kprobes are getting events for every file/folder that is affected but we care only for the entries in the cache that regard paths that we monitor this is expected to miss